### PR TITLE
Upgrade play to 2.1.5

### PR DESCRIPTION
**Pull request by maddalab**
This is a fairly long PR. It makes more sense to review the comments on each step. Unfortunately it is a all or nothing scenario. Each intermediate step will result in compilation failures.

I have completed basic testing, creation of assets, setting attributes, search for attributes etc. The dockerfile has been updated to do the same.

Currently we get a large number of compiler warnings. Primarily around catching throwable exceptions. However included amongst these are instances of Future.get() which are important.

This PR does make the situation around running all unit tests at once worse and requires the use of test-only to run individual tests. I will follow up with updates to that.

This is now done. Upgraded solr, which also requires JDK 7 , 6 isn't supported. Updated other libraries and fixed all the test (bar one).

```
[info] Passed: : Total 153, Failed 0, Errors 0, Passed 153, Skipped 0
[success] Total time: 24 s, completed Jan 31, 2015 9:54:38 PM
```

Please take a look do your testing and lets commit this.

@byxorna @Primer42 @DanSimon

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -8,7 +8,7 @@ parallelExecution in Test := false
 
 parallelExecution in IntegrationTest := false
 
-scalaVersion := "2.9.1"
```
Comment (Primer42) [inline build.sbt]: You need to update `.travis.yaml` to reflect this.
https://github.com/tumblr/collins/blob/master/.travis.yml#L7

**Comment by Primer42** [comment]
I only see compile warnings when I run this.  What compile errors were you getting @maddalab?
Since it runs OK, I would be in favor of merging, and cleaning up the warnings in a future PR, since this PR is already pretty big.

**Comment by maddalab** [comment]
updates to warning in the original comment

File: project/Build.scala [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import sbt._
 import Keys._
-import PlayProject._
+import play.Project._
 
 object ApplicationBuild extends Build {
 
     val appName         = "collins"
-    val appVersion      = "1.2.4-SNAPSHOT"
+    val appVersion      = "1.2.5"
```
Comment (byxorna) [inline project/Build.scala]: this should already be 1.3.0, and I would suggest we bump this up to 1.4.0 with the play upgrades

File: test/ApplicationSpecification.scala [hunk 3]
```diff
@@ -21,7 +21,7 @@ trait ApplicationSpecification extends mutable.Specification with ResourceFinder
       )
 
     Play.start(app)
-    evolutionFor("collins")
+    //evolutionFor("collins")
```
Comment (byxorna) [inline test/ApplicationSpecification.scala]: Is this accidentally commented out? I would expect we would want to run the evolutions for the specs.

**Comment by byxorna** [comment]
Christ... awesome work! This LGTM, especially if you make any progress on the test suite.

File: project/Build.scala [hunk 4]
```diff
@@ -1,14 +1,14 @@
 import sbt._
 import Keys._
-import PlayProject._
+import play.Project._
 
 object ApplicationBuild extends Build {
 
     val appName         = "collins"
-    val appVersion      = "1.2.4-SNAPSHOT"
```
Comment (Primer42) [inline project/Build.scala]: I can handle cutting a new release, announcements and the like, once this is merged.

**Comment by byxorna** [comment]
There are a whole mess of solr-related deprecations and such that came along with the 4.10 solr upgrade. Ill update this PR with fixes shortly.

File: app/util/concurrent/BackgroundProcessor.scala [hunk 5]
```diff
@@ -36,8 +38,8 @@ object BackgroundProcessor {
 
   type SendType[T] = Tuple2[Option[Throwable], Option[T]]
   def send[PROC_RES,RESPONSE](cmd: BackgroundProcess[PROC_RES])(result: SendType[PROC_RES] => RESPONSE)(implicit mf: Manifest[PROC_RES]) = {
-    ask(ref, cmd)(cmd.timeout).mapTo[PROC_RES].asPromise.extend1 {
-      case Redeemed(v) => result(Tuple2(None, Some(v)))
+    ask(ref, cmd)(cmd.timeout).mapTo[PROC_RES].map {
```
Comment (byxorna) [inline app/util/concurrent/BackgroundProcessor.scala]: This is broken. Try to run an asset through intake

screen shot 2015-02-01 at 5 09 11 pm

**Comment by maddalab** [comment]
snake-yaml is required. We can consider rewriting ProvisionerProfileHelper to use native yaml libraries if we have those. If you are asking "do we need ProvisionerProfileHelper?" I do not know.

File: app/util/concurrent/BackgroundProcessor.scala (hunk 5 above)
Comment (maddalab) [inline app/util/concurrent/BackgroundProcessor.scala]: Asset intake tests for IPMI you cannot test that without physical resources.

Comment (maddalab) [inline app/util/concurrent/BackgroundProcessor.scala]: Drop the exception trace from the console.

Comment (byxorna) [inline app/util/concurrent/BackgroundProcessor.scala]: No, I can run an asset through intake without IPMI working on master, just not this branch.

**Comment by byxorna** [comment]
@maddalab sorry I meant to comment on the build.sbt: https://github.com/tumblr/collins/pull/262/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceL35
It appears you are removing the snake-yaml dependency from build.sbt, but it is still in use in the ProvisionerProfileHelper. Not sure how its still compiling, but... 

Also, provisioning is broken (in addition to intake). Try to provision a fake asset yields:
```
2015-02-01 17:12:34,481 - [ERROR] - application - p.a.LoggerLike$class:error:138 -

! @6l3mm960h - Internal server error, for (POST) [/asset/tumblrtag302/provision] ->

play.api.Application$$anon$1: Execution exception[[MatchError: ProvisionerResult(TestSucceeded,Exit Code: 0, Stdout: ) (of class controllers.actors.ProvisionerResult)]]
        at play.api.Application$class.handleError(Application.scala:287) ~[play_2.10.jar:2.1.5]
        at play.api.DefaultApplication.handleError(Application.scala:381) [play_2.10.jar:2.1.5]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$play$core$server$netty$PlayDefaultUpstreamHandler$$handle$1$1.apply(PlayDefaultUpstreamHandler.scala:143) [play_2.10.jar:2.1.5]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$play$core$server$netty$PlayDefaultUpstreamHandler$$handle$1$1.apply(PlayDefaultUpstreamHandler.scala:139) [play_2.10.jar:2.1.5]
        at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play_2.10.jar:2.1.5]
        at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play_2.10.jar:2.1.5]
scala.MatchError: ProvisionerResult(TestSucceeded,Exit Code: 0, Stdout: ) (of class controllers.actors.ProvisionerResult)
        at util.concurrent.BackgroundProcessor$$anonfun$send$1.apply(BackgroundProcessor.scala:41) ~[na:na]
        at scala.util.Success$$anonfun$map$1.apply(Try.scala:206) ~[scala-library.jar:na]
        at scala.util.Try$.apply(Try.scala:161) ~[scala-library.jar:na]
        at scala.util.Success.map(Try.scala:206) ~[scala-library.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) ~[scala-library.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) ~[scala-library.jar:na]
2015-02-01 17:12:34,483 - [WARN] - application - Global$:onError:39 - Unhandled exception
play.api.Application$$anon$1: Execution exception[[MatchError: ProvisionerResult(TestSucceeded,Exit Code: 0, Stdout: ) (of class controllers.actors.ProvisionerResult)]]
        at play.api.Application$class.handleError(Application.scala:287) ~[play_2.10.jar:2.1.5]
        at play.api.DefaultApplication.handleError(Application.scala:381) [play_2.10.jar:2.1.5]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$play$core$server$netty$PlayDefaultUpstreamHandler$$handle$1$1.apply(PlayDefaultUpstreamHandler.scala:143) [play_2.10.jar:2.1.5]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$play$core$server$netty$PlayDefaultUpstreamHandler$$handle$1$1.apply(PlayDefaultUpstreamHandler.scala:139) [play_2.10.jar:2.1.5]
        at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play_2.10.jar:2.1.5]
        at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play_2.10.jar:2.1.5]
scala.MatchError: ProvisionerResult(TestSucceeded,Exit Code: 0, Stdout: ) (of class controllers.actors.ProvisionerResult)
        at util.concurrent.BackgroundProcessor$$anonfun$send$1.apply(BackgroundProcessor.scala:41) ~[na:na]
        at scala.util.Success$$anonfun$map$1.apply(Try.scala:206) ~[scala-library.jar:na]
        at scala.util.Try$.apply(Try.scala:161) ~[scala-library.jar:na]
        at scala.util.Success.map(Try.scala:206) ~[scala-library.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) ~[scala-library.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) ~[scala-library.jar:na]
```

**Comment by maddalab** [comment]
ok, snake-yaml is probably transitively pulled in as a dependency

File: app/util/concurrent/BackgroundProcessor.scala (hunk 5 above)
Comment (byxorna) [inline app/util/concurrent/BackgroundProcessor.scala]:
```2015-02-01 17:49:18,332 - [ERROR] - application - p.a.LoggerLike$class:error:138 -

! @6l3n14g43 - Internal server error, for (GET) [/resource/2] ->

play.api.Application$$anon$1: Execution exception[[MatchError: None (of class scala.None$)]]
        at play.api.Application$class.handleError(Application.scala:287) ~[play_2.10.jar:2.1.5]
        at play.api.DefaultApplication.handleError(Application.scala:381) [play_2.10.jar:2.1.5]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$play$core$server$netty$PlayDefaultUpstreamHandler$$handle$1$1.apply(PlayDefaultUpstreamHandler.scala:143) [play_2.10.jar:2.1.5]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$play$core$server$netty$PlayDefaultUpstreamHandler$$handle$1$1.apply(PlayDefaultUpstreamHandler.scala:139) [play_2.10.jar:2.1.5]
        at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play_2.10.jar:2.1.5]
        at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play_2.10.jar:2.1.5]
scala.MatchError: None (of class scala.None$)
        at util.concurrent.BackgroundProcessor$$anonfun$send$1.apply(BackgroundProcessor.scala:41) ~[na:na]
        at scala.util.Success$$anonfun$map$1.apply(Try.scala:206) ~[scala-library.jar:na]
        at scala.util.Try$.apply(Try.scala:161) ~[scala-library.jar:na]
        at scala.util.Success.map(Try.scala:206) ~[scala-library.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) ~[scala-library.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) ~[scala-library.jar:na]
2015-02-01 17:49:18,333 - [WARN] - application - Global$:onError:39 - Unhandled exception
play.api.Application$$anon$1: Execution exception[[MatchError: None (of class scala.None$)]]
        at play.api.Application$class.handleError(Application.scala:287) ~[play_2.10.jar:2.1.5]
        at play.api.DefaultApplication.handleError(Application.scala:381) [play_2.10.jar:2.1.5]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$play$core$server$netty$PlayDefaultUpstreamHandler$$handle$1$1.apply(PlayDefaultUpstreamHandler.scala:143) [play_2.10.jar:2.1.5]
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$play$core$server$netty$PlayDefaultUpstreamHandler$$handle$1$1.apply(PlayDefaultUpstreamHandler.scala:139) [play_2.10.jar:2.1.5]
        at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play_2.10.jar:2.1.5]
        at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play_2.10.jar:2.1.5]
scala.MatchError: None (of class scala.None$)
        at util.concurrent.BackgroundProcessor$$anonfun$send$1.apply(BackgroundProcessor.scala:41) ~[na:na]
        at scala.util.Success$$anonfun$map$1.apply(Try.scala:206) ~[scala-library.jar:na]
        at scala.util.Try$.apply(Try.scala:161) ~[scala-library.jar:na]
        at scala.util.Success.map(Try.scala:206) ~[scala-library.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) ~[scala-library.jar:na]
        at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:235) ~[scala-library.jar:na]```

File: app/controllers/actions/resources/IntakeStage1Action.scala [hunk 6]
```diff
@@ -3,19 +3,17 @@ package actions
...
 import util.plugins.{IpmiPowerCommand, PowerManagement}
 import util.security.SecuritySpecification
-
 import collins.power.Identify
 import collins.power.management.{PowerManagement, PowerManagementConfig}
-
 import play.api.data.Form
 import play.api.data.Forms._
 import play.api.mvc.AsyncResult
+import com.twitter.util.Await
```
Comment (nsauro) [inline app/controllers/actions/resources/IntakeStage1Action.scala]: do you want twitter's Await, or Scala's Await?

File: app/controllers/actors/ProvisionerProcessor.scala [hunk 7]
```diff
@@ -1,13 +1,14 @@
 package controllers
 package actors
 
-import akka.util.Duration
+import scala.concurrent.duration._
 import collins.provisioning.ProvisionerRequest
 import collins.shell.CommandResult
 import com.twitter.util.Future
```
Comment (nsauro) [inline app/controllers/actors/ProvisionerProcessor.scala]: I'm a bit confused to why we would be pulling in Twitter Futures/classes.  Is there any particular reason for this?

File: app/controllers/actions/resources/IntakeStage1Action.scala (hunk 6 above)
Comment (maddalab) [inline app/controllers/actions/resources/IntakeStage1Action.scala]: Twitter. Collins has used twitter finagle internally, we should probably remove it and just use akka but that is out of scope for this particular PR

**Comment by maddalab** [comment]
@byxorna fixed intake and provisioning

**Comment by byxorna** [comment]
@maddalab is this something we should be worried about?
```
[warn] Potentially incompatible versions of dependencies of {file:/Users/gabe/code/collins/}collins:
[warn]    org.scala-lang: 2.10.4, 2.10.0, 2.10.3
```

**Comment by byxorna** [comment]
@maddalab looks like provisioning and intake do indeed work :)
I think this is good from my point of view

**Comment by byxorna** [comment]
LGTM
